custom/icds_reports/reports/disha.py

```diff
@@ -29,7 +29,7 @@ def __init__(self, state_name, month):
         self.month = month
 
     def _blob_id(self):
-        return 'disha_dump-{}-{}.json'.format(self.state_name.replace(" ", ""), self.month.strftime('%Y-%m-%d'))
+        return 'disha_dump-{}-{}.json'.format(self.state_name.encode('ascii', 'ignore').replace(" ", ""), self.month.strftime('%Y-%m-%d'))
```

Comment:
This will not work on Python 3. `.encode(...)` always returns bytes, but `.replace(" ", "")` is a text (Unicode) operation, so it will fail on Python 3. Should be
```py
return 'disha_dump-{}-{}.json'.format(
    self.state_name.encode('ascii', 'ignore').decode('ascii').replace(" ", ""),
    self.month.strftime('%Y-%m-%d'),
)
```
`encode` removes non-ASCII characters while converting to bytes and `decode` converts bytes to text.

Reply:
Ha, thanks for this note! I have fixed this in a different way https://github.com/dimagi/commcare-hq/pull/22513/files#diff-48e45a26c4763ddeafc50cc7f7e97a12R34